nemo/collections/nlp/models/language_modeling/megatron/gpt_full_te_layer_autocast_spec.py

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Optional
+
+import torch
+from megatron.core import parallel_state, tensor_parallel
+from megatron.core.transformer.spec_utils import ModuleSpec
+from megatron.core.transformer.utils import make_sharded_tensors_for_checkpoint
```

Comment:
I think these just needed to be import guarded, rather than removing the import tests for every domain.

Reply:
If you want to keep the NLP import test in jenkins that's fine, but can you revert the ASR and TTS import test removal from actions?